docs/api/advanced/subscribe.mdx

```diff
@@ -35,6 +35,12 @@ subscribe(state.arr, () => console.log('state.arr has changed to', state.arr))
 state.arr.push('world')
 ```
 
-## Codesandbox demo in VanillaJS
+#### no usage tracking
+
+`subscribe` currently does not implement the fine-grained usage tracking of `useSnapshot`, so the callback will rerun anytime the subscribed proxy (or one of its child proxy) has any fields mutated, regardless of the fields accessed within the `callback`'s code.
+
+Basically, valtio's subscriptions are fired on each new version of a proxy, which aligns to exactly when new `snapshot`s need to be taken, which is separate from whether specific views on a given snapshot need to re-render.
```

Comment:
Not sure if this page is the best place for it, but this subscriptions --> on new version --> snapshots need to know when a new version happens has a nice "aha" for me, about why valtio only does fine-grained / usage-based logic "at the very end" against snapshots and not sooner in the subscription lifecycle.

Reply:
yeah, this is a nice description and we should have it somewhere, here or before.

Reply:
Agreed I like this description but feel this isn't 100% the best place for it; leaving it here for now...